packages/eas-cli/src/commands/update/index.ts

```diff
@@ -416,10 +416,34 @@ export default class UpdatePublish extends EasCommand {
         uploadedAssetCount = uploadResults.uniqueUploadedAssetCount;
         assetLimitPerUpdateGroup = uploadResults.assetLimitPerUpdateGroup;
         unsortedUpdateInfoGroups = await buildUnsortedUpdateInfoGroupAsync(assets, exp);
-        const uploadAssetSuccessMessage = uploadedAssetCount
-          ? `Uploaded ${uploadedAssetCount} ${uploadedAssetCount === 1 ? 'asset' : 'assets'}!`
-          : `Uploading assets skipped -- no new assets found!`;
-        assetSpinner.succeed(uploadAssetSuccessMessage);
+
+        // NOTE(cedric): we assume that bundles are always uploaded, and always are part of
+        // `uploadedAssetCount`, perferably we don't assume. For that, we need to refactor the
+        // `uploadAssetsAsync` and be able to determine asset type from the uploaded assets.
+        const uploadedBundleCount = uploadResults.launchAssetCount;
+        const uploadedNormalAssetCount = Math.max(0, uploadedAssetCount - uploadedBundleCount);
+        const reusedNormalAssetCount = uploadResults.uniqueAssetCount - uploadedNormalAssetCount;
+
+        assetSpinner.stop();
+        Log.withTick(
+          `Uploaded ${uploadedBundleCount} app ${uploadedBundleCount === 1 ? 'bundle' : 'bundles'}`
+        );
+        if (uploadedNormalAssetCount === 0) {
+          Log.withTick(`Uploading assets skipped - no new assets found`);
+        } else {
+          let message = `Uploaded ${uploadedNormalAssetCount} ${
+            uploadedNormalAssetCount === 1 ? 'asset' : 'assets'
+          }`;
+          if (reusedNormalAssetCount > 0) {
+            message += ` (reused ${reusedNormalAssetCount} ${
```

Comment:
I don't feel too strongly either but I'd rather see "Uploaded x unique assets" doesn't really matter how much automation we did.

Reply:
Same here, but it might throw off users. Just showing "uploaded 2 assets" and their file names seems weird. If you don't know about this reusing mechanism (I don't expect users to know about EAS internals), you are presented with "only 2 assets are uploaded".

Not feeling strongly about this though, can remove it or at least change it in what @wschurman suggests.

Reply:
I decided that I don't feel too strongly about this. Running it on my own test app the "reused" sounded fine.